R/plotreg.R

```diff
@@ -1,394 +1,372 @@
+                   "misspecified? If coefficients were renamed using the",
+                   "'custom.coef.names' argument, 'omit.coef' refers to the renamed",
+                   "coefficients."))
+    }
+    
+    
+    
+    # omit.coef
+    if (!is.na(omit.coef)) {
+        dataframe <- dataframe[!grepl(omit.coef, dataframe$co.names), ]
     }
     
-    # xlim argument
-    if (!is.null(xlim)) {
-      if (is.numeric(xlim) && length(xlim) == 2) {
-        xl <- xlim
-      } else if (length(xlim) != 2 || class(xlim) != "list") {
-        stop(paste("Horizontal limits ('xlim' argument) must be provided as", 
-            "a vector of two numerics or as a list of such vectors where each", 
-            "item corresponds to a model."))
-      } else if (class(xlim) == "list") {
-        if (length(xlim) != length(models)) {
-            stop(paste("There are", length(models), "models but", length(xlim), 
-                "xlim items."))
+    #custom coef. name
+    if (length(custom.coef.names)> 0) {
+        levels(dataframe$co.names) <- custom.coef.names
+    }
+    
+    #custom model name
+    if (length(custom.model.names)> 0) {
+        levels(dataframe$lab) <- custom.model.names
+    }
+    
+    
+    # ggplot functions   
+    if (type == "facet") {
+        
+        if (length(reorder.coef) > 0) {
+            #reorder levels 
+            dataframe$co.names <- ordered(dataframe$co.names, levels = (unique(as.character(dataframe$co.names))))
+            #reorder coef
+            reorder.coef<- rev(reorder.coef)
+            dataframe$co.names <- factor(dataframe$co.names, levels(dataframe$co.names)[reorder.coef])
+        } else {
+            #reorder levels 
+            dataframe$co.names <- ordered(dataframe$co.names, levels = (unique(as.character(dataframe$co.names))))
+            #reorder coef in original order
+            startlevel<- c(length(dataframe$co.names):1)
+            dataframe$co.names <- factor(dataframe$co.names, levels(dataframe$co.names)[startlevel])
+        }
+        
+        p <- ggplot(dataframe, aes(co.names, co)) + 
+            geom_hline(yintercept=0, 
+                       lty=2, lwd=1, 
+                       colour="grey50") +
+            geom_errorbar(aes(ymin=lower.outer, ymax=upper.outer), 
+                          lwd=1, 
+                          colour = ifelse(dataframe$signif == TRUE , signif.light, insignif.light), 
+                          width=0) +
+            geom_errorbar(aes(ymin= lower.inner, ymax= upper.inner), 
+                          lwd=2.5, 
+                          colour = ifelse(dataframe$signif == TRUE, signif.medium, insignif.medium), 
+                          width=0) +
+            geom_point(size=3, 
+                       pch = ifelse(dataframe$signif == TRUE, 21, 22), 
+                       fill = ifelse(dataframe$signif == TRUE, signif.dark, insignif.dark)) +
+            coord_flip() + 
+            theme_bw() +
```

Comment:
With any plot customizations like these (e.g., themes, titles, ...), you risk less flexibility on the part of the user. I would suggest leaving these choices to the user. Most R users should be familiar more or less with `ggplot2` customization, so I think allowing them to change things like the theme is more important than giving them nice defaults.

But I am not actually sure if it's impossible to change these defaults because you are setting them explicitly or if it's rather because you print the plot at the end rather than just returning it. Can you try that out and then play with something like this (based on the regression model from the `lm` help file)?

```r
plotreg(list(lm.D9, lm.D90)) + theme_dark() + ggtitle("my title")
```
I would also recommend adding an example like this to the help file.